### PR TITLE
[PlacementGroup]Add PlacementGroup wait java api

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
Add placement group `wait` java api:
```
 boolean wait(int timeoutMs)
```
Simplify the process of judging whether PG is created successfully, and the logic is implemented in c++ layer, which can unify the API of Java and python.

## Related issue number
https://github.com/ray-project/ray/issues/12617

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: java/runtime/src/main/java/io/ray/runtime/gcs/GlobalStateAccessor.java [hunk 1]
```diff
@@ -82,8 +82,7 @@ private void validateGlobalStateAccessorPointer() {
 
   public byte[] getPlacementGroupInfo(PlacementGroupId placementGroupId) {
     synchronized (GlobalStateAccessor.class) {
-      Preconditions.checkNotNull(placementGroupId,
-          "PlacementGroupId can't be null when get placement group info.");
+      validateGlobalStateAccessorPointer();
```
Comment (clay4444) [inline java/runtime/src/main/java/io/ray/runtime/gcs/GlobalStateAccessor.java:85]: Good catch 👍

File: src/ray/gcs/gcs_server/gcs_placement_group_manager.cc [hunk 2]
```diff
@@ -388,6 +400,32 @@ void GcsPlacementGroupManager::HandleGetAllPlacementGroup(
...
+      PlacementGroupID::FromBinary(request.placement_group_id());
+  RAY_LOG(DEBUG) << "Waiting for placement group until ready, placement group id = "
+                 << placement_group_id;
+
+  // If the placement group is not exist, return directly.
+  if (!registered_placement_groups_.contains(placement_group_id)) {
+    GCS_RPC_SEND_REPLY(send_reply_callback, reply,
+                       Status::Invalid("Placement group is not exist."));
+  }
+
```
Comment (clay4444) [inline src/ray/gcs/gcs_server/gcs_placement_group_manager.cc:417]: maybe we should check whether the state is already created from `registered_placement_groups_ `? if not, the callback maybe never invoked.

Comment (ffbin) [reply src/ray/gcs/gcs_server/gcs_placement_group_manager.cc:417]: Yeah, you are right, thanks.

**Review by raulchen — APPROVED** [review]

File: src/ray/core_worker/lib/java/io_ray_runtime_task_NativeTaskSubmitter.cc [hunk 3]
```diff
@@ -293,6 +293,16 @@ Java_io_ray_runtime_task_NativeTaskSubmitter_nativeRemovePlacementGroup(
   THROW_EXCEPTION_AND_RETURN_IF_NOT_OK(env, status, (void)0);
 }
 
+JNIEXPORT jboolean JNICALL
+Java_io_ray_runtime_task_NativeTaskSubmitter_nativeWaitPlacementGroupReady(
+    JNIEnv *env, jclass p, jbyteArray placement_group_id_bytes, jint timeout_ms) {
+  const auto placement_group_id =
+      JavaByteArrayToId<ray::PlacementGroupID>(env, placement_group_id_bytes);
+  auto status = ray::CoreWorkerProcess::GetCoreWorker().WaitPlacementGroupReady(
```
Comment (raulchen) [inline src/ray/core_worker/lib/java/io_ray_runtime_task_NativeTaskSubmitter.cc:301]: throw an exception to Java if status is not ok.

Comment (ffbin) [reply src/ray/core_worker/lib/java/io_ray_runtime_task_NativeTaskSubmitter.cc:301]: fixed

File: src/ray/protobuf/gcs_service.proto [hunk 4]
```diff
@@ -527,20 +527,28 @@ message GetAllPlacementGroupReply {
...
   // Get placement group information via gcs service.
   rpc GetPlacementGroup(GetPlacementGroupRequest) returns (GetPlacementGroupReply);
-
   // Get information of all placement group from GCS Service.
   rpc GetAllPlacementGroup(GetAllPlacementGroupRequest)
       returns (GetAllPlacementGroupReply);
+  // Wait for placement group until ready.
+  rpc WaitPlacementGroupUntilReady(WaitPlacementGroupUntilReadyRequest)
+      returns (WaitPlacementGroupUntilReadyReply);
 }
```
Comment (rkooo567) [inline src/ray/protobuf/gcs_service.proto:554]: Newline

**Review by rkooo567 — DISMISSED** [review]
LGTM, but there's newline issue. Please fix it and lmk!

**Comment by rkooo567** [comment]
Also @ffbin Can you create an issue with milestone to implement the python API?

**Comment by ffbin** [comment]
> Also @ffbin Can you create an issue with milestone to implement the python API?

I have create an issue, thanks.

**Comment by ffbin** [comment]
> LGTM, but there's newline issue. Please fix it and lmk!
hi @rkooo567 , i have fix it and pls help take a look, thanks.

**Pull request merged**